userdocs/diagnostics/expression-too-complex.md

```diff
@@ -0,0 +1,16 @@
+# Type Inference in Complex Expressions
+For some complex expressions, the Swift compiler is unable to infer types in reasonable time. This can occur for a number of reasons, but some common ones include:
+
+- The expression makes use of many overloaded operators and/or functions.
+- The expression contains literals which can be used as values of many different types (e.g. integer and floating point literals).
+- The expression includes a collection literal with a large number of nontrivial element expressions.
```

Comment:
What does "nontrivial" mean here? We use "trivial" sometimes to designate certain POD types, but I'd imagine that's not what you're talking about in this situation.

Reply:
I think `notrivial` could be removed here because it really depends on number of elements rather than on their complexity in `large collection literals`.